.github/workflows/ci.yml

```diff
@@ -64,6 +64,50 @@ jobs:
         run: |
           ./scripts/check-usage-of-log-warn.sh
 
+  audit:
+    name: cargo audit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set-Up
+        run: |
+          sudo apt-get update
+          sudo apt install -y protobuf-compiler
+
+      - name: Configure rustc version
+        run: |
+          source ci/env
+          echo "RUSTC_VERSION=$RUSTC_VERSION" >> $GITHUB_ENV
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.RUSTC_VERSION }}
+          target: wasm32-unknown-unknown
+          profile: minimal
+          override: true
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Install cargo audit
+        run: |
+          cargo install cargo-audit --features=fix
+
+      - name: Run cargo audit
+        run: |
+          cargo audit --ignore RUSTSEC-2020-0071
+          CC_MATCH=$(cargo audit --ignore RUSTSEC-2020-0071 --color never | egrep "^.── creditcoin" || true)
+
+          # fail for anything which is a 1st level dependency of Creditcoin
+          if [ -n "$CC_MATCH" ]; then
+              echo "FAIL: Direct dependencies of Creditcoin found in audit report"
+              echo "INFO: Inspect the output above for more information"
+              exit 1
+          else
+              echo "INFO: No direct dependencies of Creditcoin found in audit report"
+              exit 0
+          fi
```

Comment:
I second Ada here, not sure I like ignoring indirect dependencies. I feel like if an advisory is really unactionable we can just add it to the whitelist case by case.

Reply:
> not sure I like ignoring indirect dependencies.

We are not ignoring dependencies, on the contrary.  

The first check will fail regardless where the dependency comes from if there is a vulnerability (excluding vulnerabilities which we explicitly decided to ignore). 

The second check will fail even for warnings if they are reported against 1st level dependencies of Creditcoin.

Reply:
Ahh okay I see